### PR TITLE
fixed #342: ambiguous comparison operators (c++20)

### LINKED ISSUE #342: C++20: ambiguous overloads for operator==
When building with Clang 10 in C++20 mode, a lot of comparison operators are now ambiguous.

Example:

    /home/daniel/git/sqlpp11/tests/Interpret.cpp:100:21: error: use of overloaded operator '==' is ambiguous (with operand types 'const sqlpp::column_t<test::TabBar, test::TabBar_::Alpha>' and 'parameter_t<value_type_of<sqlpp::column_t<test::TabBar, test::TabBar_::Alpha> >, sqlpp::column_t<test::TabBar, test::TabBar_::Alpha> >' (aka 'parameter_t<sqlpp::integral, sqlpp::column_t<test::TabBar, test::TabBar_::Alpha> >'))
      serialize(t.alpha == parameter(t.alpha), printer).str();
                ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~
    /home/daniel/git/sqlpp11/include/sqlpp11/basic_expression_operators.h:127:10: note: candidate function [with T = sqlpp::parameter_t<sqlpp::integral, sqlpp::column_t<test::TabBar, test::TabBar_::Alpha> >]
        auto operator==(T t) const -> _new_binary_expression_t<equal_to_t, T>
             ^
    /home/daniel/git/sqlpp11/include/sqlpp11/basic_expression_operators.h:127:10: note: candidate function (with reversed parameter order) [with T = sqlpp::column_t<test::TabBar, test::TabBar_::Alpha>]

Any ideas how to prevent this / work around this issue?
Thanks!

### PR DESCRIPTION
**Pull request by dermojo**
see #342

### CODE REVIEW

File: include/sqlpp11/column.h [hunk 1]
```diff
@@ -63,6 +63,10 @@ namespace sqlpp
     template <typename T>
     using _is_valid_assignment_operand = is_valid_assignment_operand<value_type_of<ColumnSpec>, T>;
 
+    // disambiguation for C++20
```
Comment (rbock) [inline include/sqlpp11/column.h:66]: Since this (still) looks like a clang bug, can you link to the bug report?

**Comment by rbock** [comment]
Thanks!
We might have to expect other places to fail in the same way for clang with C++20 (anything that inherits from `expression_operators`, depending on how quickly the bug in clang will be fixed.

**Comment by dermojo** [comment]
Thank you (for the library and taking the time for this issue - I'm curious about sqlpp17 ;-) )
Our tests so far were fine, no other issues popping up. I also built sqlpp11's tests with clang-10 in C++20 mode, and it ran fine (ignoring other warnings...). This might be all...

**Comment by rbock** [comment]
Thanks for the feedback.

To mention it here, too: According to Richard Smith, the observed ambiguity is not a bug in clang, but indeed a result of changes in C++20 (the committee is thinking of ways to alleviate the situation).

Regarding sqlpp17: The code is pretty advanced already, but I am not sure if I will finish it. C++20 is so much nicer with concepts and modules. That being said: The real thrill will be C++ with reflection and generation, whenever that will be available.

**Pull request merged**